Crashes when no "disconnection" handler is registered

If you do not register a handler with the "disconnection" event, there is runtime error:

```
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4dbe1d]

goroutine 11 [running]:
github.com/googollee/go-socket%2eio.(*decoder).Close(0x0)
	~/.gvm/pkgsets/go1.4.2/global/src/github.com/googollee/go-socket.io/parser.go:161 +0x7d
github.com/googollee/go-socket%2eio.(*socketHandler).onPacket(0xc20800c860, 0x0, 0xc20824d200, 0x0, 0x0, 0x0, 0x0, 0x0)
	~/.gvm/pkgsets/go1.4.2/global/src/github.com/googollee/go-socket.io/handler.go:146 +0x11f
github.com/googollee/go-socket%2eio.func·004()
	~/.gvm/pkgsets/go1.4.2/global/src/github.com/googollee/go-socket.io/socket.go:117 +0xb8
github.com/googollee/go-socket%2eio.(*decoder).Close(0xc208033040)
	~/.gvm/pkgsets/go1.4.2/global/src/github.com/googollee/go-socket.io/parser.go:161 +0x3b
github.com/googollee/go-socket%2eio.(*socketHandler).onPacket(0xc20800c860, 0xc208033040, 0xc208033080, 0x0, 0x0, 0x0, 0x0, 0x0)
	~/.gvm/pkgsets/go1.4.2/global/src/github.com/googollee/go-socket.io/handler.go:146 +0x11f
github.com/googollee/go-socket%2eio.(*socket).loop(0xc20803b320, 0x0, 0x0)
	~/.gvm/pkgsets/go1.4.2/global/src/github.com/googollee/go-socket.io/socket.go:135 +0x37b
github.com/googollee/go-socket%2eio.func·003(0xc20803b320)
	~/.gvm/pkgsets/go1.4.2/global/src/github.com/googollee/go-socket.io/server.go:93 +0x28
created by github.com/googollee/go-socket%2eio.(*Server).loop
	~/.gvm/pkgsets/go1.4.2/global/src/github.com/googollee/go-socket.io/server.go:94 +0x109
```

A workaround is to register an empty handler:

```
  server.On("connection", func(so socketio.Socket) {
    so.On("disconnection", func() {  })
  })
```

--- a/parser.go
+++ b/parser.go
@@ -158,9 +158,11 @@ func newDecoder(r frameReader) *decoder {
 }
 
 func (d *decoder) Close() {
-	d.currentCloser.Close()
-	d.current = nil
-	d.currentCloser = nil
+	if d != nil && d.currentCloser != nil {
+		d.currentCloser.Close()
+		d.current = nil
+		d.currentCloser = nil
+	}
 }
 
 func (d *decoder) Decode(v *packet) error {

